eng/versioning.targets

```diff
@@ -0,0 +1,190 @@
+    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName($(SyncInfoFile)))" />
+    <WriteLinesToFile
+      Condition="'$(SourceRevisionId)' != ''"
+      File="$(SyncInfoFile)"
+      Lines="$(SourceRevisionId)"
+      Overwrite="true" />
+  </Target>
+
+  <!--
+    GenerateNativeVersionFile target is a standalone target intended to be pulled into a build once as
+    a pre-step before kicking off a native build. It will generate a _version.h or _version.c depending
+    on the OS it is targeting.
+  -->
+  <Target Name="GenerateNativeVersionFile"
+          DependsOnTargets="_InitializeAssemblyVersion;InitializeSourceControlInformationFromSourceControlManager">
+
+    <Error Condition="'$(SourceRevisionId)' == ''" Text="SourceRevisionId is not set, which means the SourceLink targets are not included in the build. Those are needed to produce a correct sha for our build outputs." />
+
+    <PropertyGroup>
+      <_SourceBuildInfo> %40Commit: $(SourceRevisionId)</_SourceBuildInfo>
+    </PropertyGroup>
+
+   <PropertyGroup Condition="'$(OS)' == 'Windows_NT'">
+      <NativeVersionFile Condition="'$(NativeVersionFile)' == ''">$(ArtifactsObjDir)_version.h</NativeVersionFile>
+      <_WindowsFileVersion>$(FileVersion.Replace('.', ','))</_WindowsFileVersion>
+      <_Windows_VER_DEBUG>0</_Windows_VER_DEBUG>
+      <_Windows_VER_DEBUG Condition="'$(Configuration)'=='Debug'">VS_FF_DEBUG</_Windows_VER_DEBUG>
+
+      <_NativeVersionFileContents>
+<![CDATA[
+#ifndef VER_COMPANYNAME_STR
+#define VER_COMPANYNAME_STR         "Microsoft Corporation"
+#endif
+#ifndef VER_FILEDESCRIPTION_STR
+#define VER_FILEDESCRIPTION_STR     "$(AssemblyName)"
+#endif
+#ifndef VER_INTERNALNAME_STR
+#define VER_INTERNALNAME_STR        VER_FILEDESCRIPTION_STR
+#endif
+#ifndef VER_ORIGINALFILENAME_STR
+#define VER_ORIGINALFILENAME_STR    VER_FILEDESCRIPTION_STR
+#endif
+#ifndef VER_PRODUCTNAME_STR
+#define VER_PRODUCTNAME_STR         "Microsoft\xae .NET Framework"
+#endif
+#undef VER_PRODUCTVERSION
+#define VER_PRODUCTVERSION          $(_WindowsFileVersion)
+#undef VER_PRODUCTVERSION_STR
+#define VER_PRODUCTVERSION_STR      "$(_WindowsFileVersion)$(_SourceBuildInfo)"
+#undef VER_FILEVERSION
+#define VER_FILEVERSION             $(_WindowsFileVersion)
+#undef VER_FILEVERSION_STR
+#define VER_FILEVERSION_STR         "$(_WindowsFileVersion)$(_SourceBuildInfo)"
+#ifndef VER_LEGALCOPYRIGHT_STR
+#define VER_LEGALCOPYRIGHT_STR      "\xa9 Microsoft Corporation. All rights reserved."
+#endif
+#ifndef VER_DEBUG
+#define VER_DEBUG                   $(_Windows_VER_DEBUG)
+#endif
+]]>
+      </_NativeVersionFileContents>
+    </PropertyGroup>
+
+    <!--
+      Copy the NativeVersion.rc file next to the version header so that it can be picked
+      up and used in the native build along with the version.h file.
+    -->
+    <Copy SourceFiles="$(MSBuildThisFileDirectory)..\Tools\NativeVersion.rc"
```

Comment:
This seems to still be coming from buildtools, are we planning on moving it somewhere else? Should we generate this file on the fly just like version.c/h?